rhino/src/main/java/org/mozilla/javascript/IdScriptableObject.java

```diff
@@ -936,6 +942,21 @@ protected ScriptableObject getOwnPropertyDescriptor(Context cx, Object id) {
         return desc;
     }
 
+    /**
+     * Overridden in the base class for different descriptors
+     *
+     * @return ScriptableObject
+     */
+    ScriptableObject buildDataDescriptorHelper(
```

Comment:
please add @Override... same for the next and maybe you can also copy the comment

Reply:
Removed these as they don't actually solve the problem with name and length attributes.